astropy/time/formats.py

```diff
@@ -2175,6 +2176,204 @@ def value(self):
+        ((?P<hr>{re_float}) \s* hr \s*)?
+        ((?P<min>{re_float}) \s* min \s*)?
+        ((?P<s>{re_float}) \s* s)?
+        \s* $
+        """,
+        re.VERBOSE,
+    )
+
+    def _check_val_type(self, val1, val2):
+        if val1.dtype.kind not in ("S", "U") and val1.size:
+            raise TypeError(f"Input values for {self.name} class must be strings")
+        if val2 is not None:
+            raise ValueError(
+                f"{self.name} objects do not accept a val2 but you provided {val2}"
+            )
+        return val1, None
+
+    def parse_string(self, timestr):
+        """Read time from a single string"""
+        # Datetime components required for conversion to JD by ERFA, along
+        # with the default values.
+        components = ("yr", "d", "hr", "min", "s")
+
+        if (match := self.re_ydhms.match(timestr)) is None:
+            raise ValueError(
+                f"Time delta '{timestr}' does not match {self.name} format"
+            )
+
+        tm = match.groupdict()
+        vals = [float(tm[component] or 0.0) for component in components]
+        if tm["sign"] == "-":
+            vals = [-val for val in vals]
+
+        return vals
+
+    def set_jds(self, val1, val2):
+        """Parse the time strings contained in val1 and get jd1, jd2."""
+        # Be liberal in what we accept: convert bytes to ascii.
+        # Here .item() is needed for arrays with entries of unequal length,
+        # to strip trailing 0 bytes.
+        to_string = (
+            str if val1.dtype.kind == "U" else lambda x: str(x.item(), encoding="ascii")
+        )
+        iterator = np.nditer(
+            [val1, None, None, None, None, None],
+            flags=["zerosize_ok"],
+            op_dtypes=[None] + 5 * [np.double],
+        )
+        for val, yr, day, hr, min, sec in iterator:
+            val = to_string(val)
+            (
+                yr[...],
+                day[...],
+                hr[...],
+                min[...],
+                sec[...],
+            ) = self.parse_string(val)
+
+        yrs, days, hrs, mins, secs = iterator.operands[1:]
+        jd1 = yrs * 365.25 + days + hrs / 24.0 + mins / 1440.0 + secs / 86400.0
+        self.jd1, self.jd2 = day_frac(jd1, 0.0)
+
+    def to_value(self, parent=None, out_subfmt=None):
+        out_subfmt = out_subfmt or self.out_subfmt
+        subfmt = self._get_allowed_subfmt(out_subfmt)
+
+        iterator = np.nditer(
+            [self.jd1, self.jd2, None],
+            flags=["refs_ok", "zerosize_ok"],
+            op_dtypes=[None, None, object],
+        )
+
+        for jd1, jd2, out in iterator:
+            jd1_, jd2_ = day_frac(jd1, jd2)  # Why is this required?
+
+            # Naive and low-precision implementation for now
+            jd = jd1_ + jd2_
+            sign = "-" if jd < 0 else ""
+            jd = np.abs(jd)
+
+            if subfmt in ["*", "multi"]:
+                comps = self.get_multi_comps(jd)
```

Comment:
I think one might as well keep precision here too, so `self.get_multi_comps(jd1_, jd2_)`

That also means above
```
if jd < 0:
    jd1_ = -jd1_
    jd2_ = -jd2_
    jd = np.abs(jd)
    sign = "-"
else:
    sign = ""
```

Reply:
Done.